packages/engine/paima-sm/src/cde-cardano-mint-burn.ts

```diff
@@ -0,0 +1,31 @@
+import { ENV } from '@paima/utils';
+import { cdeCardanoMintBurnInsert, createScheduledData } from '@paima/db';
+import type { SQLUpdate } from '@paima/db';
+import type { CdeCardanoMintBurnDatum } from './types.js';
+
+export default async function processDatum(
+  cdeDatum: CdeCardanoMintBurnDatum
+): Promise<SQLUpdate[]> {
+  const cdeId = cdeDatum.cdeId;
+  const prefix = cdeDatum.scheduledPrefix;
+  const txId = cdeDatum.payload.txId;
+  const assets = JSON.stringify(cdeDatum.payload.assets);
+  const metadata = cdeDatum.payload.metadata || undefined;
+
+  const scheduledBlockHeight = Math.max(cdeDatum.blockNumber, ENV.SM_START_BLOCKHEIGHT + 1);
+  const scheduledInputData = `${prefix}|${txId}|${metadata}|${assets}`;
```

Comment:
what happens if `metadata` contains a `|` in it? Is that possible in the encoding we use?

Reply:
it's not really possible because right now it's just hex (of the cbor format). If we decode it to a json I guess it should be possible. Do we want to do that?

Reply:
I think hex is fine 👍